mmdet/models/detectors/kd_one_stage.py

```diff
@@ -0,0 +1,93 @@
+import torch
+
+from ..builder import DETECTORS
+from .single_stage import SingleStageDetector
+
+
+@DETECTORS.register_module()
+class KnowledgeDistillationSingleStageDetector(SingleStageDetector):
+    r"""Implementation of `Distilling the Knowledge in a Neural Network.
+    <https://arxiv.org/abs/1503.02531>`_.
+    """
+
+    def __init__(self,
+                 backbone,
+                 neck,
+                 bbox_head,
+                 teacher_config,
+                 teacher_ckpt=None,
+                 eval_teacher=True,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super().__init__(backbone, neck, bbox_head, train_cfg, test_cfg,
+                         pretrained)
+
+        self.eval_teacher = eval_teacher
+
+        from mmdet.apis.inference import init_detector
```

Comment:
I still wonder why `init_detector` can only be imported inside the class initialization.

Reply:
https://stackoverflow.com/questions/9252543/importerror-cannot-import-name-x
I found someone has met a similar problem. init-detector is not defined when import kd_one_stage.

Reply:
Then we'd better use `build_detector` and `load_checkpoint` explicitly to avoid circular importing.